pages/beacon.tsx

```diff
@@ -68,14 +70,22 @@ const Beacon = () => {
 
     const data = decodeData((searchParams.get("data") ?? code) as string);
 
+    if (
+      data.appUrl === "http://localhost:3000" ||
+      data.appUrl.includes("tzsafe")
```

Comment:
also need to check the `data.name` is "TzSafe" as the first checking condition. 
`http://localhost:3000` should be removed

Reply:
I don't think we should check the name. Anyone can put the name he wants, checking the `appUrl` is better imo

Reply:
but if the dapp's name  is "TzSafe", it's high chance that it's malicious one.
Usually, you don't use the existed name as your product name.